aiohttp/client_reqrep.py

```diff
@@ -606,17 +614,19 @@ async def send(self, conn: "Connection") -> "ClientResponse":
         # - not CONNECT proxy must send absolute form URI
         # - most common is origin form URI
         if self.method == hdrs.METH_CONNECT:
-            connect_host = self.url.raw_host
-            assert connect_host is not None
-            if helpers.is_ipv6_address(connect_host):
-                connect_host = f"[{connect_host}]"
+            if _YARL_SUPPORTS_HOST_SUBCOMPONENT:
+                connect_host = self.url.host_subcomponent
+                assert connect_host is not None
+            else:
+                connect_host = self.url.raw_host
+                assert connect_host is not None
+                if helpers.is_ipv6_address(connect_host):
+                    connect_host = f"[{connect_host}]"
```

Comment:
https://app.codecov.io/gh/aio-libs/aiohttp/blob/master/aiohttp%2Fclient_reqrep.py#L611

This missing coverage is existing, but will soon disappear after https://github.com/aio-libs/aiohttp/pull/9305 and a followup to remove this branch